tiledb/sm/filesystem/azure.cc

```diff
@@ -649,20 +751,211 @@ Status Azure::touch(const URI& uri) const {
   return Status::Ok();
 }
 
-Status Azure::write(const URI& uri, const void* buffer, uint64_t length) {
+Status Azure::write(
+    const URI& uri, const void* const buffer, const uint64_t length) {
+  if (!uri.is_azure()) {
+    return LOG_STATUS(Status::AzureError(
+        std::string("URI is not an Azure URI: " + uri.to_string())));
+  }
+
+  Buffer* const write_cache_buffer = get_write_cache_buffer(uri.to_string());
+
+  uint64_t nbytes_filled;
+  RETURN_NOT_OK(
+      fill_write_cache(write_cache_buffer, buffer, length, &nbytes_filled));
+
+  if (!use_block_list_upload_) {
```

Comment:
why do we check this after the `fill_write_cache` call here but not below?

Reply:
When the user configures either S3/Azure to not use chunked uploads, they may call ::write() as many times as they want but the cumulative size of the writes must not exceed the in-memory write cache. This `if` statement checks that they do not exceed that size. The `return` on line 774 ensures this non-chunked upload path does not reach the next `fill_write_cache`.